helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -888,6 +885,91 @@ public Response updateCloudConfig(@PathParam("clusterId") String clusterId,
     return OK();
   }
 
+  /**
+   * Get the global target task thread pool size of the cluster, a value that's used to construct
+   * task thread pools for the cluster's instances and is created by users. This value is used
+   * only if the instances don't specify their own target thread pool sizes in their
+   * InstanceConfigs; if this value needs to be used but is not specified, a default value -
+   * TaskConstants.DEFAULT_TASK_THREAD_POOL_SIZE will be used instead.
+   * @param clusterId - the cluster id of which the global target thread pool size belongs to
+   */
+  @GET
+  @Path("{clusterId}/global-target-task-thread-pool-size")
+  public Response getGlobalTargetTaskThreadPoolSize(@PathParam("clusterId") String clusterId) {
+    ClusterConfig config;
+    try {
+      config = getClusterConfigWithConfigAccessor(clusterId);
+    } catch (Exception ex) {
+      LOG.error("Failed to get cluster config for cluster {}. Exception: {}", clusterId, ex);
+      return serverError(ex);
+    }
+    if (config == null) {
+      return notFound();
+    }
+    int threadPoolSize = config.getGlobalTargetTaskThreadPoolSize();
+    if (threadPoolSize == TaskConstants.TARGET_TASK_THREAD_POOL_SIZE_NOT_SET) {
+      return notFound();
+    }
+    Map<String, Integer> resultMap = new HashMap<>();
+    resultMap.put(ClusterProperties.globalTargetTaskThreadPoolSize.name(), threadPoolSize);
+    return JSONRepresentation(resultMap);
+  }
+
+  /**
+   * Update the global target task thread pool size of the cluster. The global target task thread
+   * pool size goes to ClusterConfig, and is applied to all instances of the cluster for task thread
+   * pool construction. If an instance doesn't specify its target thread pool size in
+   * InstanceConfig, then this value in ClusterConfig will be used to construct its task thread
+   * pool. The newly-set target task thread pool size will take effect upon a JVM restart. If none
+   * of the global and per-instance target thread pool sizes are set, a default value -
+   * TaskConstants.DEFAULT_TASK_THREAD_POOL_SIZE will be used.
+   * @param clusterId - the cluster id of which the global target thread pool size is set to
+   * @param threadPoolSize - the thread pool size to be set
+   */
+  @POST
+  @Path("{clusterId}/global-target-task-thread-pool-size/{threadPoolSize}")
```

Comment:
To make it more generic, how about

`"{clusterId}/task-framework/thread-pool-size/"`? @jiajunwang @dasahcc  what do you think?

This way, we open up a namespace for other future task framework-related configs.

Also, should we use queryParam instead because thread pool size is not a resource?

Reply:
Ok to queryParam. I will leave the `task-framework` namespace discussion open for further participation.